engines/rbd.c

```diff
@@ -127,6 +135,7 @@ static int _fio_setup_rbd_data(struct thread_data *td,
 
 }
 
+#define EVENT_SOCKET_TYPE_EVENTFD 2 /* add for rbd poll */
```

Comment:
enum {                                                                          
  EVENT_TYPE_PIPE = 1,                                                          
  EVENT_TYPE_EVENTFD = 2                                                        
}; 

So it is there. I'd propose adding a configure check for whether set_image_type(..., EVENT_TYPE_EVENTFD) works or not, and use it in rbd.c if that is the case.